nav2_tasks/include/nav2_tasks/bt_action_node.hpp

```diff
@@ -165,7 +171,10 @@ class BtActionNode : public BT::AsyncActionNode
     if (status() == BT::NodeStatus::RUNNING) {
       action_client_->async_cancel_goal(goal_handle_);
       auto future_cancel = action_client_->async_cancel_goal(goal_handle_);
-      rclcpp::spin_until_future_complete(node_, future_cancel);
+      {
```

Comment:
Semantics, but this is right before leaving the scope of the `if`, so the additional scoping for the lock isn't required